receiver/awscontainerinsightreceiver/cadvisor/cadvisor_linux.go

```diff
@@ -0,0 +1,35 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package cadvisor
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/host"
+)
+
+type Cadvisor struct {
+}
+
+func New(containerOrchestrator string, machineInfo *host.MachineInfo, logger *zap.Logger) *Cadvisor {
+	return &Cadvisor{}
+}
+
+func (c *Cadvisor) GetMetrics() []pdata.Metrics {
```

Comment:
To introduce PRs gradually, I think going package by package is better than outside-in skeleton like this, it's confusing reviewing unimplemented code I think.

Reply:
I am not sure that I can do package by package now because the dependencies are not there yet. For example, this PR needs to be merged first: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3211. (Could you review it as well?)

Another concern is that the PR for each package will be big as you can see from here: https://github.com/aws-observability/aws-otel-collector/tree/main/internal/receiver/awscontainerinsightreceiver. I initially just want to set up the skeleton so that I can fill in the details piece by piece (and also package by package). The design and implementation details are documented in the `README.md` file. However if you prefer to review package by package first, I can definitely change the PR accordingly :)

Reply:
I guess the difficulty in reviewing is, for example, it's unclear if this `return nil` seems like a bug or is going to be fixed later. Usually we'd start by declaring the component + config, then implement it in parts. package by package was a rough term, but I meant code by code, maybe per-file, rather than outside-in which causes this problem. Does it seem possible?

Reply:
As discussed offline, I will add TODO comments in the code to better explain the intent of each functions.